### PR TITLE
Add Communication endpoint. Update status

### PR DESCRIPTION
**Pull request by rubenelshof**
Added Communication proto & buffer. Updated solaredge.py & readme to add communication.
Reformatted Status proto file. Updated Status proto & buffer to add Communication information.

Added some Todo items as I am not getting that information when I use curl to get the raw response.

### CODE REVIEW

**Comment by rubenelshof** [comment]
Just found a issue I didnt have when using curl.

When using a script like this (retrieves data and dumps it as json):
```
#!/usr/bin/env python3
import datetime
import time
import json
from solaredge_local import SolarEdge
from google.protobuf.json_format import MessageToDict

se_client = SolarEdge("http://192.168.2.37");

def get_test_data():
    try:
        status = se_client.get_communication()
        return MessageToDict(status)
    except Exception as e:
        print('Failed to get test status from SolarEdge inverter: ', e)
        return

def construct_test_data(data):
    test_data = {
        "test": data,
    }
    return test_data

def display_test_data():
    try:
        se_test_construct = construct_test_data(get_test_data())
        print(json.dumps(se_test_construct, indent=4, sort_keys=True))
    except KeyError as e:
        pass

display_test_data()
```
I get the following error:
```
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    from solaredge_local import SolarEdge
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\solaredge_local\__init__.py", line 1, in <module>
    from solaredge_local.solaredge import SolarEdge
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\solaredge_local\solaredge.py", line 6, in <module>
    from .communication_pb2 import Communication
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\solaredge_local\communication_pb2.py", line 1877
    _sym_db.RegisterMessage(Communication.Rs4851.Protocol.None)
                                                             ^
SyntaxError: invalid syntax
```

Dont know why.
Anybody got any clues?

**Comment by dlmcpaul** [comment]
Is it possible that "None" is a reserved word in python?

File: message_types/communication.proto [hunk 1]
```diff
@@ -0,0 +1,169 @@
+syntax = "proto3";
+
+message Communication {
+    message ServerChannel {
+        message Lan {
+            bool value = 1;
+        }
+        // Todo add wifi rs & zigbee. Unclear which values.
```
Comment (ashb) [inline message_types/communication.proto]: Wifi is 3

Comment (ashb) [reply message_types/communication.proto]:
```serverChannel {
  lan {
  }
  3 {
    1: 1
  }
  4: ""
  6: ""
}```

Is what I get when I decode this with my inverter set to WiFi mode

File: message_types/communication.proto [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        Lan lan = 1;
+    }
+
+    message Lan {
+        message Dhcp {
+            bool value = 1;
```
Comment (ashb) [inline message_types/communication.proto]:
```suggestion
            bool enabled = 1;
```

File: message_types/status.proto [hunk 3]
```diff
@@ -1,128 +1,162 @@
...
+                bool sunspec = 4;
+                bool none = 5;
+            }
+            Protocol protocol = 3;
+        }
+        
+        //Todo: add more information about modbus and slaves
+        Rs4851 rs4851 = 1;
+        Rs4852 rs4852 = 2;
+        int32 lan = 3;
```
Comment (ashb) [inline message_types/status.proto]:
```
communication {
  rs4851 {
    protocol {
      seSlave: true
    }
  }
  rs4852 {
    protocol {
      seSlave: true
    }
  }
  5 {
    1: "my-wifi-ssid"
    2: 4
  }
}
```

```protobuf
    message WiFi {
        string ssid = 1;
        // int32 ??
    }

    ...
    WiFi wifi = 5;
```

File: message_types/status.proto [hunk 4]
```diff
@@ -1,128 +1,162 @@
...
+    message Inverters {
+        message Unit {
+            string dspSn = 1;
+            float voltage = 2;
+            OptimizersStatus optimizersStatus = 3;
+            Temperature temperature = 4;
+            int32 fan = 5;
+            string errorCode = 6;
+            int32 subSystem = 7;
+            string badPosition = 8;
```
Comment (ashb) [inline message_types/status.proto]: I also get this in my response:

```
    9 {
      1 {
        1: 5000
        2: 100
      }
      2 {
        1: 1100000
        2: 100
      }
      3: 10
    }
    10 {
      1: 236143
      2: 100
    }
```

9.2.1 (`1100000`) there equals the value shown in `#/commissioning/maintenance/diagnostics/isolation-status` - a value of `11000 kOhm`

I don't know what the other values are.

Comment (ashb) [reply message_types/status.proto]: 9.2.2 might be the "no fault found"

File: message_types/status.proto (hunk 3 above)
Comment (ashb) [reply message_types/status.proto]: 5.2 is a signal strength indicator. 4 seems to be 100%/full bars.

File: message_types/communication.proto (hunk 1 above)
Comment (rubenelshof) [reply message_types/communication.proto]: I have added Wifi and changed "value" to "enabled".

File: message_types/communication.proto (hunk 2 above)
Comment (rubenelshof) [reply message_types/communication.proto]: I have changed that now.

File: message_types/status.proto (hunk 3 above)
Comment (rubenelshof) [reply message_types/status.proto]: Added. Thank you for reporting it.

File: message_types/status.proto (hunk 4 above)
Comment (rubenelshof) [reply message_types/status.proto]: 9 is for isolation.
9.1 is for "Alpha" which I dont know what it stands for.
9.2.2 is scaling of isolation.
9.3 is fault location.

I dont know what 10.1 could be.

I have added isolation to the status proto.

Thank you for reporting this and the other points.

**Comment by rubenelshof** [comment]
@ashb 
Could you check the latest updates?
And could you tell me what you get with decoding Communication and than number 6?
For example I get:
6 {
  3: 3 Signal strength
  4 { wpsDuration
    1: 60 wpsDuration value
  }
  5 { extAntenna
    1: 0 extAntenna value.
  }
}

In this list I am missing the ssid and a list of discovered networks.

**Comment by ashb** [comment]
It turns out the webconsole prints out the decoded information!

```
wifi: {
  chosenNetwork: {
    name: { value: "milkyway" }
    password: undefined
    passwordNeeded: { value: false }
    saved: { value: true }
    secured: { value: true }
    signalStrength: 4
  discoverdNetworksList: []
  extAntenna: { value: true }
  hgEnable: undefined
  hgStatus: undefined
  status: 1
  wpsDuration: { value: 60 }
}
```

I don't know how to map those values in to an order though.

**Comment by ashb** [comment]
Oh - we can un-minify the JS, and pull out the `proto.web_communication.Communication.Wifi.deserializeBinaryFromReader` and related functions etc.

**Pull request closed without merging**